test/functional/feature_anchors.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test Anchors functionality"""
+
+import os
+
+from test_framework.p2p import P2PInterface
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+
+def check_node_connections(*, node, num_in, num_out):
+    info = node.getnetworkinfo()
+    assert_equal(info["connections_in"], num_in)
+    assert_equal(info["connections_out"], num_out)
+
+
+class AnchorsTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def setup_network(self):
+        self.setup_nodes()
+
+    def run_test(self):
+        self.log.info("Add 2 block-relay-only connections to node 0")
+        for i in range(2):
+            self.log.debug(f"block-relay-only: {i}")
+            self.nodes[0].add_outbound_p2p_connection(
+                P2PInterface(), p2p_idx=i, connection_type="block-relay-only"
+            )
+
+        self.log.info("Add 5 inbound connections to node 0")
+        for i in range(5):
+            self.log.debug(f"inbound: {i}")
+            self.nodes[0].add_p2p_connection(P2PInterface())
+
+        self.log.info("Check node 0 connections")
+        check_node_connections(node=self.nodes[0], num_in=5, num_out=2)
+
+        # 127.0.0.1
+        ip = "7f000001"
+
+        # Since the ip is always 127.0.0.1 for this case, we store only the port to identify the peers
+        block_relay_nodes_port = []
+        inbound_nodes_port = []
+        for p in self.nodes[0].getpeerinfo():
+            addr_split = p["addr"].split(":")
+            if p["connection_type"] == "block-relay-only":
+                block_relay_nodes_port.append(hex(int(addr_split[1]))[2:])
+            else:
+                inbound_nodes_port.append(hex(int(addr_split[1]))[2:])
+
+        self.log.info("Stop node 0")
+        self.stop_node(0)
+
+        node0_anchors_path = os.path.join(
+            self.nodes[0].datadir, "regtest", "anchors.dat"
+        )
+
+        self.log.info("Check anchors.dat file in the node directory")
+        assert os.path.exists(node0_anchors_path)
+
+        # It should contain only the block-relay-only addresses
+        self.log.info("Check the addresses in anchors.dat")
+
+        anchors = ""
```

Comment:
wouldn't it make more sense for this to be an array?

Reply:
I don't think so. I am concatenating the lines to verify if the addresses are into this. I don't think an array would be useful here.

Reply:
Let's assume ip_port is `aa` and anchors is `['fa', 'af']`

With array:

```
>>> 'aa' in ['fa', 'af']
False
```

With string

```
>>> 'aa' in 'faaf'
True
```

Am I missing something?

Reply:
I'm using string for the reason you're showing with your example. I think the question about using array or string is: how is anchors.dat structured?

Let's suppose:

ip_port is 7f0000013e01 and the file content is:
fabfb5da02fc5a030000e1f505000000000000000000000000000000000000ffff7f0000013e01fc5a030000e1f505000000000000000000000000000000000000ffff7f0000013e0075f7d7cf33e71e42f8add50c305f091fb881b6fa5633932559d86cd9f6141260

but we have:
fabfb5da02fc5a030000e1f505000000000000000000000000000000000000ffff`7f00` (breaks line)
`00013e`01fc5a030000e1f505000000000000000000000000000000000000ffff7f000 (breaks line)
0013e0075f7d7cf33e71e42f8add50c305f091fb881b6fa5633932559d86cd9f6141260 (breaks line)

in this case, my logic using string would be useful. 
But how is the file structure for this case?

Anyway, almost 100% of my tests showed me that anchors.dat usually has just one line.

Reply:
anchors.dat is binary data, so why would newlines matter and what is the point of `rstrip`?

Reply:
I can remove `rstrip` I guess, it doesn't affect anything, thanks.

Reply:
So, can I read only the first line?
 I was using `array` but the test `was failing, assuming:
```
item (first line): fabfb5da02fc5a030000e1f505000000000000000000000000000000000000ffff7f0000012b6efc5a030000e1f505000000000000000000000000000000000000ffff7f0000012b6d1c665ef1c1d383c4869170e021306a945bdbfb9f39c12bb2e0
ip_port: 7f0000012b6e
ip_port in item: True
ip_port: 7f0000012b6d
ip_port in item: True
item (second line): 279f548ecaf3
ip_port: 7f0000012b6e
ip_port in item: False`
```

Reply:
If newlines don't have a meaning, they shouldn't be interpreted. You can just use `read()`?

Reply:
Yes, just pushed the changes. It makes sense. Could you please check?